src/org/omegat/core/machinetranslators/ApertiumTranslate.java

```diff
@@ -138,4 +159,63 @@ protected String getJsonResults(String json) {
 
         return tr;
     }
+
+    /**
+     * Whether or not to use a custom Apertium server
+     */
+    private boolean useCustomServer() {
+        String value = System.getProperty(PROPERTY_APERTIUM_SERVER_CUSTOM,
+                Preferences.getPreference(PROPERTY_APERTIUM_SERVER_CUSTOM));
+        return Boolean.parseBoolean(value);
+    }
+
+    @Override
+    public boolean isConfigurable() {
+        return true;
+    }
+
+    @Override
+    public void showConfigurationUI(Window parent) {
+
+        JCheckBox apiCheckBox = new JCheckBox(OStrings.getString("APERTIUM_CUSTOM_SERVER_LABEL"));
+        apiCheckBox.setSelected(useCustomServer());
+
+        MTConfigDialog dialog = new MTConfigDialog(parent, getName()) {
+            @Override
+            protected void onConfirm() {
+                boolean temporary = panel.temporaryCheckBox.isSelected();
+                System.setProperty(PROPERTY_APERTIUM_SERVER_CUSTOM, Boolean.toString(apiCheckBox.isSelected()));
+                Preferences.setPreference(PROPERTY_APERTIUM_SERVER_CUSTOM, apiCheckBox.isSelected());
+                String server = panel.valueField1.getText().trim();
+                String apiKey = panel.valueField2.getText().trim();
+                setCredential(PROPERTY_APERTIUM_SERVER_URL, server, temporary);
+                setCredential(PROPERTY_APERTIUM_SERVER_KEY, apiKey, temporary);
+            }
+        };
+
+        ItemListener toggleInterface = new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent event) {
+                dialog.panel.valueLabel1.setEnabled(apiCheckBox.isSelected());
+                dialog.panel.valueLabel2.setEnabled(apiCheckBox.isSelected());
+                dialog.panel.valueField1.setEnabled(apiCheckBox.isSelected());
+                dialog.panel.valueField2.setEnabled(apiCheckBox.isSelected());
+                dialog.panel.temporaryCheckBox.setEnabled(apiCheckBox.isSelected());
+            }
+        };
+
+        apiCheckBox.addItemListener(toggleInterface);
+
+        dialog.panel.itemsPanel.add(apiCheckBox,1);
+        dialog.panel.valueLabel1.setText(OStrings.getString("APERTIUM_CUSTOM_SERVER_URL_LABEL"));
+        dialog.panel.valueField1.setText(getCredential(PROPERTY_APERTIUM_SERVER_URL));
+        dialog.panel.valueField1.setColumns(20);
+        dialog.panel.valueLabel2.setText(OStrings.getString("APERTIUM_CUSTOM_SERVER_KEY_LABEL"));
+        dialog.panel.valueField2.setText(getCredential(PROPERTY_APERTIUM_SERVER_KEY));
```

Comment:
The `temporaryCheckBox` needs to be properly initialized. See `isCredentialStoredTemporarily` in `BaseTranslate`.

Reply:
Fixed in latest commit, thanks for pointing this out.